### PR TITLE
feat: update webhook documentation with more grant types

### PR DESCRIPTION
**Pull request by sgal**
## Related Issue or Design Document

Documentation update for https://github.com/ory/hydra/pull/3427.

This change adds guides for adding/updating custom token claims via webhook feature for the following set of grant types:
- `authorization_code`
- `client_credentials`
- `refresh_token`
- `urn:ietf:params:oauth:client-assertion-type:jwt-bearer`

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

**Comment by sgal** [comment]
Updated docs page at Preview - https://docs-rnke3r89h-ory.vercel.app/docs/hydra/guides/custom-token-claims

**Review by tomekpapiernik — CHANGES_REQUESTED** [review]

File: docs/hydra/guides/custom-token-claims.mdx [hunk 1]
```diff
@@ -1,33 +1,51 @@
...
 
-To enable the token refresh webhook, configure the `oauth2.refresh_token_hook` key in the Hydra configuration:
+### Grant types
+
+The hooks feature is enabled for the following grant types:
+
+- `authorization_code`
+- `client_credentials`
+- `refresh_token`
+- `urn:ietf:params:oauth:client-assertion-type:jwt-bearer` - see [RFC7523](https://www.rfc-editor.org/rfc/rfc7523)
```
[the remark this answers is not in the document]
Comment (sgal) [reply docs/hydra/guides/custom-token-claims.mdx:28]: Makes sense to shorten the name 👍 It is an OAuth2 grant, see https://www.rfc-editor.org/rfc/rfc7523.

**Comment by sgal** [comment]
@tomekpapiernik Thanks for the review. I reverted the document ID and the sidebar changes and fixed the writing. Please have a look.

**Comment by tomekpapiernik** [comment]
Thanks for updating your PR @sgal ! I took the liberty of editing this a bit more. Most important change - I updated the Ory CLI to use --add instead of --replace, as the keys you want to work with aren't present in the oauth2 config from the get-go - they must be added. The --replace command doesn't do that. I'll approve this PR and it can be merged along with the code PR you opened.

**Review by tomekpapiernik — DISMISSED** [review]

**Comment by sgal** [comment]
@tomekpapiernik Thanks a lot for the review and fixes 👏

**Review by hperl — APPROVED** [review]
LGTM, thanks for your contribution @sgal!

**Pull request merged**